packages/opentelemetry-sampler-aws-xray/src/remote-sampler.ts

```diff
@@ -58,9 +70,46 @@ export class AWSXRayRemoteSampler implements Sampler {
     this.startRulePoller();
   }
 
-  shouldSample(): SamplingResult {
-    // Implementation to be added
-    return { decision: SamplingDecision.NOT_RECORD };
+  shouldSample(
+    context: Context,
+    traceId: string,
+    spanName: string,
+    spanKind: SpanKind,
+    attributes: Attributes,
+    links: Link[]
+  ): SamplingResult {
+    if (this._ruleCache.isExpired()) {
+      // go to fallback sampler .shouldSample
+      return this._fallBackSampler.shouldSample(
+        context,
+        traceId,
+        spanName,
+        spanKind,
+        attributes,
+        links
+      );
+    }
+
+    const matchedRule = this._ruleCache.getMatchedRule(
+      attributes,
+      this._resource
+    );
+
+    // TODO: update after verifying if default rule will always match,
+    // this means that this method will always return return { decision: matchedRule.sample(attributes) }
+    // as long as the rule cache has not expired.
+    if (matchedRule) {
+      return { decision: matchedRule.sample(attributes) };
+    }
+
+    return this._fallBackSampler.shouldSample(
```

Comment:
IIUC we will end up here in case even the default rule is not matched, likely due to some bug? If so can we add a log statement here to be aware of it?

Reply:
Good point, I'll add a warning log here to help with debugging